### PR TITLE
Add support to auto dl nightly simc

### PR DESCRIPTION
**Pull request by psykzz**
### Description of the Change
This PR adds support for automatically downloading the latest nightly build of simc from http://downloads.simulationcraft.org/nightly/?C=M;O=D.

Inspiration was taken from the approach at https://github.com/SimCMinMax/AutoSimC/blob/master/main.py but I've updated it a bit to look a little cleaner and split into more functions.

This requires both `--local` and `--auto_download` to be specified to work.

Additionally, this fixed a build issue with the latest version of NumPy on windows that can't be verified the "fix" was to pin to the previous version (which i did here) though that did cause some issues for Linux. 
Let me know if you want me to remove/keep that in.

I formatted the additional code with `black`, but avoided touching existing code in the file.

### Benefits

This just simplifies getting simc, and automates the process.

### Applicable Issues

Context on the NumPy issue can be found here
https://developercommunity.visualstudio.com/comments/1242128/view.html (linked to the solution)

### CODE REVIEW

File: internal/simc.py [hunk 1]
```diff
@@ -19,4 +25,107 @@ def sim_local(simc_path, profile_location, output_location, iterations):
...
+        files = glob.glob(f"{download_dir}/simc*")
+        for f in files:
+            if (
+                os.path.basename(f) != current_dir
+                and os.path.basename(f) != f"{current_dir}.7z"
+            ):
+                print(f"Removing old simc from '{os.path.basename(f)}'.")
+                os.remove(f)
+    except:
+        print(
```
Comment (psykzz) [inline internal/simc.py]: This fails unless you run the script as admin, which i didn't want to start asking people to do.
For now just telling them to manually clean it up.

File: sim.py [hunk 2]
```diff
@@ -142,10 +142,20 @@ def main():
...
         '--local', help='indicate if the simulation should run local.', action='store_true')
+    parser.add_argument(
+        '--auto_download', help='indicate if we should automatically download latest simc.', action='store_true')
     args = parser.parse_args()
 
     sys.path.insert(0, args.dir)
 
+
+    # Download simc if needed
+    if(args.local and args.auto_download):
```
Comment (seanpeters86) [inline sim.py]: make sure and apply proper pep format here, this should have a space. 
```suggestion
    if (args.local and args.auto_download):
```

File: sim.py [hunk 3]
```diff
@@ -142,10 +142,20 @@ def main():
...
+        from internal.simc import download_latest
+        from local_secrets import simc_path
+        simc_path['latest'] = download_latest()
```
Comment (seanpeters86) [inline sim.py:156]: can we move `download_latest()` and some of the helper functions over to an `internal/` script? This file is already kind of bloated so if we can keep things as separate as possible that would be helpful from a logical standpoint.

**Review by seanpeters86 — CHANGES_REQUESTED** [review]

File: sim.py (hunk 3 above)
Comment (psykzz) [reply sim.py:156]: Updated this to pull from internal.auto_download instead.

**Review by seanpeters86 — APPROVED** [review]

**Pull request merged**